### ISSUE TITLE
Comment before `--ifdef` results in unexpected modifier indentation

### ISSUE THREAD

**Issue by mikemahony** [nicklockwood/SwiftFormat#1527]
When you have a comment before an `#if` the indentation of any chained modifier is unexpected.

### Input
```swift
struct ContentView: View {
    var body: some View {

        // swiftformat:options --ifdef indent

        Text("Hello, world!")
        // Comment above
        #if os(macOS)
            .padding()
        #endif

        Text("Hello, world!")
        #if os(macOS)
            // Comment inside
            .padding()
        #endif

        // swiftformat:options --ifdef no-indent

        Text("Hello, world!")
        // Comment above
        #if os(macOS)
            .padding()
        #endif

        Text("Hello, world!")
        #if os(macOS)
            // Comment inside
            .padding()
        #endif

        // swiftformat:options --ifdef outdent

        Text("Hello, world!")
        // Comment above
        #if os(macOS)
            .padding()
        #endif

        Text("Hello, world!")
        #if os(macOS)
            // Comment inside
            .padding()
        #endif
    }
}
```

### Configuration
All three `--ifdef` options.

### Expected output
```swift
struct ContentView: View {
    var body: some View {

        // swiftformat:options --ifdef indent

        Text("Hello, world!")
        // Comment above
        #if os(macOS)
            .padding()
        #endif

        Text("Hello, world!")
        #if os(macOS)
            // Comment inside
            .padding()
        #endif

        // swiftformat:options --ifdef no-indent

        Text("Hello, world!")
        // Comment above
        #if os(macOS)
            .padding()
        #endif

        Text("Hello, world!")
        #if os(macOS)
            // Comment inside
            .padding()
        #endif

        // swiftformat:options --ifdef outdent

        Text("Hello, world!")
// Comment above
#if os(macOS)
            .padding()
#endif

        Text("Hello, world!")
#if os(macOS)
            // Comment inside
            .padding()
#endif
    }
}
```

### Actual output
```swift
struct ContentView: View {
    var body: some View {

        // swiftformat:options --ifdef indent

        Text("Hello, world!")
        // Comment above
        #if os(macOS)
        .padding()
        #endif

        Text("Hello, world!")
        #if os(macOS)
            // Comment inside
            .padding()
        #endif

        // swiftformat:options --ifdef no-indent

        Text("Hello, world!")
        // Comment above
        #if os(macOS)
        .padding()
        #endif

        Text("Hello, world!")
        #if os(macOS)
            // Comment inside
            .padding()
        #endif

        // swiftformat:options --ifdef outdent

        Text("Hello, world!")
// Comment above
#if os(macOS)
.padding()
#endif

        Text("Hello, world!")
#if os(macOS)
            // Comment inside
            .padding()
#endif
    }
}
```

**Comment by nicklockwood** [comment]
@mikemahony thanks for reporting this - I've added a fix for the next release.

**Comment by nicklockwood** [comment]
@mikemahony fixed in 0.52.5
